test/blackbox/common/DDSBlackboxTestsDataRepresentationQos.cpp

```diff
@@ -0,0 +1,156 @@
+
+        // Deserialize encapsulation.
+        deser.read_encapsulation();
+
+        last_encoding = deser.get_encoding_flag();
+
+        return HelloWorldPubSubType::deserialize(payload, data);
+    }
+
+    static eprosima::fastcdr::EncodingAlgorithmFlag last_encoding;
+
+    static DataRepresentationId_t last_data_representation;
+};
+
+eprosima::fastcdr::EncodingAlgorithmFlag
+MockHelloWorldPubSubType::last_encoding {eprosima::fastcdr::EncodingAlgorithmFlag::PL_CDR};
+
+DataRepresentationId_t MockHelloWorldPubSubType::last_data_representation { XML_DATA_REPRESENTATION};
+
+using DataRepresentationQosVector = std::vector<DataRepresentationId_t>;
+
+class DataRepresentationQosCompatibility : public ::testing::TestWithParam<std::tuple<
+                DataRepresentationQosVector,
+                DataRepresentationQosVector,
+                bool
+                >>
+{
+};
+
+
+DataRepresentationQosVector EMPTY_VECTOR {};
+DataRepresentationQosVector XCDR_VECTOR {XCDR_DATA_REPRESENTATION};
+DataRepresentationQosVector XCDR2_VECTOR {XCDR2_DATA_REPRESENTATION};
+DataRepresentationQosVector BOTH_XCDR_VECTOR {XCDR_DATA_REPRESENTATION, XCDR2_DATA_REPRESENTATION};
+
+TEST_P(DataRepresentationQosCompatibility, check_compatibility)
+{
+    PubSubWriter<MockHelloWorldPubSubType> writer(TEST_TOPIC_NAME);
+    PubSubReader<MockHelloWorldPubSubType> reader(TEST_TOPIC_NAME);
+
+    writer.data_representation(std::get<0>(GetParam())).init();
+    reader.reliability(RELIABLE_RELIABILITY_QOS).data_representation(std::get<1>(GetParam())).init();
+
+    ASSERT_TRUE(reader.isInitialized());
+    ASSERT_TRUE(writer.isInitialized());
+
+    if (std::get<2>(GetParam()))
+    {
+        writer.wait_discovery();
+        reader.wait_discovery();
+
+        auto data = default_helloworld_data_generator();
+
+        reader.startReception(data);
+        // Send data
+        writer.send(data);
+        // In this test all data should be sent.
+        ASSERT_TRUE(data.empty());
+        // Block reader until reception finished or timeout.
+        reader.block_for_all();
+
+        MockHelloWorldPubSubType* writer_type =
+                dynamic_cast<MockHelloWorldPubSubType*>(writer.get_type_support().get());
+        MockHelloWorldPubSubType* reader_type =
+                dynamic_cast<MockHelloWorldPubSubType*>(reader.get_type_support().get());
+        if (std::get<0>(GetParam()).empty() || XCDR_DATA_REPRESENTATION == std::get<0>(GetParam()).at(0))
+        {
+            ASSERT_EQ(XCDR_DATA_REPRESENTATION, writer_type->last_data_representation);
+            ASSERT_EQ(eprosima::fastcdr::EncodingAlgorithmFlag::PLAIN_CDR, reader_type->last_encoding);
+        }
+        else if (XCDR2_DATA_REPRESENTATION == std::get<0>(GetParam()).at(0))
+        {
+            ASSERT_EQ(XCDR2_DATA_REPRESENTATION, writer_type->last_data_representation);
+            ASSERT_EQ(eprosima::fastcdr::EncodingAlgorithmFlag::PLAIN_CDR2, reader_type->last_encoding);
+        }
```

Comment:
We could add the expected values to the test parameters, thus simplifying these lines.

Reply:
Done in bbf761728